Code/Tools/AssetProcessor/native/unittests/AssetScannerUnitTests.cpp

```diff
@@ -7,82 +7,81 @@
  */
 
 #include "AssetScannerUnitTests.h"
+#include <AzCore/std/chrono/clocks.h>
+#include <AzTest/Utils.h>
 #include <native/AssetManager/assetScanner.h>
 #include <native/utilities/PlatformConfiguration.h>
 #include <native/unittests/UnitTestRunner.h> // for CreateDummyFile
 #include <QApplication>
 #include <QElapsedTimer>
-#include <QSharedPointer>
 #include <QTemporaryDir>
 
 namespace UnitTest
 {
     TEST_F(AssetScannerUnitTest, AssetScanner_ScanMultipleFolders_ExpectedFilesAndFoldersFound)
     {
         using namespace AssetProcessor;
-        QSharedPointer<QCoreApplication> m_qApp;
+        AZStd::unique_ptr<QCoreApplication> m_qApp;
 
-        // QApplications require command line arguments, so create something to meet that requirement.
-        int argC{ 1 };
-        char commandLineBuffer[AZ_MAX_PATH_LEN];
-        char* commandLineBufferAddress{ commandLineBuffer };
-        azstrcpy(commandLineBuffer, AZ_ARRAY_SIZE(commandLineBuffer), "no_argv_supplied");
+        // Qt documentation claims QCoreApplication's constructor requires a greater than zero argC
+        // and a valid argV, however the version of Qt in use works fine with 0 and nullptr.
+        int argC = 0;
+        m_qApp.reset(new QApplication(argC, nullptr));
 
-        m_qApp.reset(new QApplication(argC, &commandLineBufferAddress));
+        AZ::Test::ScopedAutoTempDirectory tempEngineRoot;
+        //const char* tempPath(tempEngineRoot.GetDirectory());
+        //QDir tempPath(tempEngineRoot.path());
```

Comment:
You can remove the commented out code now.

Reply:
Fixed